RecoCTPPS/PixelLocal/src/RPixPlaneCombinatoryTracking.cc

```diff
@@ -358,6 +359,119 @@ void RPixPlaneCombinatoryTracking::findTracks(){
+      }
+    }
+  }; // map< shiftedPeriod, map<DetID,shiftScheme> >
+  const auto& shiftStatusInitialRun = std::prev(isPlaneShifted.upper_bound(run));
+  unsigned short shiftedROC = 10;
+  CTPPSPixelIndices pixelIndices;
+
+  // Selecting the shifted ROC
+  if(romanPotId_.arm() == 0) shiftedROC = 0;
+  if(romanPotId_.arm() == 1) shiftedROC = 5;
+
+  // Loop over found tracks to set recoInfo_
+  for(auto & track : localTrackVector_){
+    if(romanPotId_ != rpId_arm0_st2 && romanPotId_ != rpId_arm1_st2){
+      track.setRecoInfo(ReconstructionInfo::notShiftedRun);
+      if(verbosity_>=2) edm::LogInfo("RPixPlaneCombinatoryTracking")<<"Analyzing run: "<<run
+        <<"\nTrack belongs to Arm "<<romanPotId_.arm()<<" Station "
+        <<romanPotId_.station();
+
+      continue;
+    }
+    unsigned short bxShiftedPlanesUsed = 0;
+    unsigned short bxNonShiftedPlanesUsed = 0;
+    unsigned short hitInShiftedROC = 0;
+
+    auto const& fittedHits = track.getHits();
+    auto const& planeFlags = (shiftStatusInitialRun->second).at(romanPotId_);
+
+    for(const auto & planeHits : fittedHits){
+      unsigned short plane = CTPPSPixelDetId(planeHits.detId()).plane();
+      for(const auto & hit : planeHits){
+        if(hit.getIsUsedForFit()){
+          if(pixelIndices.getROCId(hit.minPixelCol(),hit.minPixelRow()) == shiftedROC) hitInShiftedROC++; // Count how many hits are in the shifted ROC
+          if(planeFlags.at(plane)) bxShiftedPlanesUsed++; // Count how many bx-shifted planes are used
+          else if(planeFlags != std::vector<bool>(6,false)) bxNonShiftedPlanesUsed++; // Count how many non-bx-shifted planes are used, only if there are shifted planes 
+        }
+      }
+    }
+
+    // Set recoInfo_ value
+    track.setRecoInfo(ReconstructionInfo::invalid); // Initially setting it as invalid. It has to match one of the following options.
+    if(hitInShiftedROC < 3) track.setRecoInfo(ReconstructionInfo::notShiftedRun);
+    else{
+      if(bxShiftedPlanesUsed == 0 && bxNonShiftedPlanesUsed == 0) track.setRecoInfo(ReconstructionInfo::notShiftedRun);    // Default value for runs without bx-shift
+      if(bxShiftedPlanesUsed == 3 && bxNonShiftedPlanesUsed == 0) track.setRecoInfo(ReconstructionInfo::allShiftedPlanes); // Track reconstructed in a shifted ROC, only with bx-shifted planes
+      if(bxShiftedPlanesUsed == 0 && bxNonShiftedPlanesUsed == 3) track.setRecoInfo(ReconstructionInfo::noShiftedPlanes);  // Track reconstructed in a shifted ROC, only with non-bx-shifted planes
+      if(bxShiftedPlanesUsed >  0 && bxNonShiftedPlanesUsed >  0) track.setRecoInfo(ReconstructionInfo::mixedPlanes);      // Track reconstructed in a shifted ROC, with mixed planes
+    }
+    if(bxShiftedPlanesUsed + bxNonShiftedPlanesUsed > 6){
+      edm::LogError("RPixPlaneCombinatoryTracking")<< "Error in RPixPlaneCombinatoryTracking::findTracks -> " << "More than 6 points found for a track, skipping.";
```

Comment:
how often does this happen?
Is the other track data coming from RPixPlaneCombinatoryTracking still usable in this condition?
If the track can still be used in analysis, this is at best a LogInfo.

Reply:
This should never happen. I'm setting it as exception.